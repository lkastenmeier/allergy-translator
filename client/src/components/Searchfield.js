import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const InputField = styled.input`
  display: flex;
  text-align-last: center;
  width: 50%;
  min-height: 50px;
  margin: 10px auto 5px auto;
  background-color: ${props => props.theme.light};
  border: 2px solid ${props => props.theme.highlight};
  color: ${props => props.theme.text};
  font-size: 1rem;
  input:active {
    border-color: transparent;
    border-bottom: 2px solid red;
  }
`;

export default function SearchField({ input }) {
  function onLocationInput(event) {
    const { value } = event.target;
    input(value);
  }
  return <InputField placeholder="Where are you?" onChange={onLocationInput} />;
}

SearchField.propTypes = {
  onChange: PropTypes.func
};
