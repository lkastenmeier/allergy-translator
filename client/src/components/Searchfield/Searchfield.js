//Searchfield to specify userlocation on HelpPage

//Dependencies
import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const StyledSearchfield = styled.input`
  display: flex;
  text-align-last: center;
  width: 95%;
  min-height: 50px;
  margin: 10px auto 5px auto;
  background-color: ${props => props.theme.light};
  border: 2px solid ${props => props.theme.highlight};
  color: ${props => props.theme.text};
  font-size: 1rem;
  @media (min-width: 500px) {
    width: 300px;
  }
`;

export default function Searchfield({ input, placeholder }) {
  function onLocationInput(event) {
    //extract value and pass it to map via Help.js
    const { value } = event.target;
    input(value);
  }
  return (
    <StyledSearchfield
      data-testid="searchfield"
      placeholder={placeholder}
      onChange={onLocationInput}
    />
  );
}

Searchfield.propTypes = {
  input: PropTypes.func,
  placeholder: PropTypes.string
};
