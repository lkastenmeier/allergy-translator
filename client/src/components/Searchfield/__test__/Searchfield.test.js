import React from "react";
import ReactDOM from "react-dom";
import Searchfield from "./../Searchfield";
import "@testing-library/jest-dom/extend-expect";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Searchfield />, div);
});

it("renders searchfield correctly", () => {
  const { getByTestId } = render(<Searchfield placeholder="test" />);
  expect(getByTestId("searchfield").placeholder).toEqual("test");
});

it("matches snapshot", () => {
  const tree = renderer.create(<Searchfield />).toJSON();
  expect(tree).toMatchSnapshot();
});
