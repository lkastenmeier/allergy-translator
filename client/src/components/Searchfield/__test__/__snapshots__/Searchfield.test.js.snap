// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<input
  className="sc-bdVaJa kiitDh"
  data-testid="searchfield"
  onChange={[Function]}
/>
`;
